monai/utils/misc.py

```diff
@@ -57,11 +64,11 @@ def ensure_tuple_rep(tup, dim):
     """
     Returns a copy of `tup` with `dim` values by either shortened or duplicated input.
     """
-    tup = ensure_tuple(tup)
-    if len(tup) >= dim:
-        return tup[:dim]
-    if len(tup) == 1:
-        return tup * dim
+    if not issequenceiterable(tup):
+        return (tup,) * dim
```

Comment:
This function is a common utility which is not only used in transforms, so the input `tup` can be objects instead of just `int`, `float` or `string`. `(tup,) * dim` is a **shallow copy** for objects which may cause an error later:
```py
>>> a = {"data": 1}
>>> b = (a,) * 5
>>> print(b)
[{'data': 1}, {'data': 1}, {'data': 1}, {'data': 1}, {'data': 1}]
>>> b[0]['data'] = 2
>>> print(b)
[{'data': 2}, {'data': 2}, {'data': 2}, {'data': 2}, {'data': 2}]
```
What do you think?

Thanks.

Reply:
The current master branch:

```
from misc import *
In [8]: a={"data":1}                                                                                                                           
In [9]: b=ensure_tuple_rep(a,1)                                                                                                                
In [10]: b                                                                                                                                     
Out[10]: ('data')
```

New code:
```
from misc import *
a={"data":1}
b=ensure_tuple_rep(a,1)
Out[6]: ('data',)
```

Both cases produce the same results with your example, and both are probably wrong results.  This seems like a bug that exists, but is out of scope for the intent of this PR.  

A more complete usecase would need to be identified for fixing the separate issue that is identified.

Reply:
Hi @wyli ,

As you added the original `ensure_tuple_rep` function, do I miss any special case here?
Thanks.

Reply:
my understanding is that `ensure_tuple_rep` is for user input validation, duplicating the items is the expected behaviour, for example:
```python
>>> from monai.utils import ensure_tuple_rep
>>> a = [[1]]
>>> param = ensure_tuple_rep(a, 3)
>>> param
([1], [1], [1])
>>> param[0][0] = 2
>>> param
([2], [2], [2])
```
if the user wants  three different input params, they should specify 
`a = [[1], [2], [3]]`. `ensure_tuple_rep([[1], [2], [3]], 3)` won't change it